### PR TITLE
Feature/dashboard

### PR DESCRIPTION
**Pull request by MatHenriquez**
# Summary
- Create the **Dashboard component**.
- Add  a **Header component** in the Dashboard.
- Add **new account helpers** in the *helpers folder*.

#Details
- Add new ***Header component*** to display the wallet brand, public key and sign out button.
- Add a **Dashboard** to show the user his balance in Lumens.

# Evidences

## Dashboard of a funded account
funded-dashboard

## Dashboard of an unfunded account
unfunded_dashboard

### CODE REVIEW

File: app/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  setIsLogged: (value: boolean) => void;
+}> = ({ publicKey, isLogged, setIsLogged }) => {
+  const [linkText, setLinkText] = useState("copy");
+
+  const handleCopyText: (text: string) => void = (text): void => {
+    navigator.clipboard.writeText(text);
+    setLinkText("copied!");
+    setTimeout(() => {
+      setLinkText("copy");
+    }, 1000);
```
Comment (Genebson) [inline app/components/Header.tsx]: Save this magic number inside a variable named `const DELAY_IN_MILLISECONDS = 1000;`

File: app/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+const UserKey: FC<{
+  publicKey: string;
+  handleCopyText: (text: string) => void;
+  linkText: string;
+}> = ({ publicKey, handleCopyText, linkText }) => {
+  return (
+    <div className="flex">
+      <p className="text-sm font-bold mr-2">
+        {publicKey.slice(0, 5) + "..." + publicKey.slice(-2)}
```
Comment (Genebson) [inline app/components/Header.tsx]: Save this in a function, please.

File: app/dashboard/page.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+"use client";
+import React, { FC, useEffect, useState } from "react";
+import Navbar from "../components/Header";
+import accountHelper from "../helpers/account";
+import { Account } from "stellar-sdk";
+
+const Dashboard: FC = () => {
+  const [isFunded, setIsFunded] = useState(false);
+  const [publicKey, setPublicKey] = useState("" as string);
+  const [isLogged, setIsLogged] = useState(publicKey !== "");
```
Comment (Genebson) [inline app/dashboard/page.tsx]: Why not just start with `true` as an initial value?

File: app/helpers/account.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { Server } from "stellar-sdk";
+const server: Server = new Server("https://horizon-testnet.stellar.org");
+
+const loadAccount = async (publicKey: string) => {
+  try {
+    const account = await server.loadAccount(publicKey);
+    return account;
```
Comment (Genebson) [inline app/helpers/account.ts]:
```suggestion
    return await server.loadAccount(publicKey);
```

File: app/helpers/account.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    const account = await loadAccount(publicKey);
+    balance = account.balances.find(
+      (balance) => balance.asset_type === "native"
+    )?.balance;
+  } catch (error) {
+    throw Error("Unfunded account");
+  }
+
+  return balance;
+};
```
Comment (Genebson) [inline app/helpers/account.ts]: This doesn't work?
```suggestion
const getBalance = async (publicKey: string): Promise<string | undefined> => {
  try {
    const account = await loadAccount(publicKey);
    return account.balances.find(
      (balance) => balance.asset_type === "native"
    )?.balance;
  } catch (error) {
    throw Error("Unfunded account");
  }
};
```

File: app/components/Header.tsx (hunk 1 above)
Comment (MatHenriquez) [reply app/components/Header.tsx]: I have initialized a constant with this value and used it place of the literal

File: app/components/Header.tsx (hunk 2 above)
Comment (MatHenriquez) [reply app/components/Header.tsx]: I have created a function called *abbreviatePublicKey* and implemented this logic in it

File: app/dashboard/page.tsx (hunk 3 above)
Comment (MatHenriquez) [reply app/dashboard/page.tsx]: I changed the default value of isFunded to true.

File: app/helpers/account.ts (hunk 4 above)
Comment (MatHenriquez) [reply app/helpers/account.ts]: I have removed the intermediate constant and now the function returns the method call directly.

File: app/helpers/account.ts (hunk 5 above)
Comment (MatHenriquez) [reply app/helpers/account.ts]: I have changed this function to avoid using unnecessary variables as you have pointed out.

**Review by Genebson — APPROVED** [review]

**Pull request merged**